cscs-checks/microbenchmarks/cpu/likwid/likwid.py

```diff
@@ -8,85 +8,85 @@
-                          'memory': '1200MB'},
-            'dom:mc':    {'L1': '288kB', 'L2': '2304kB', 'L3': '23MB',
-                          'memory': '1800MB'},
-            'dom:gpu':   {'L1': '192kB', 'L2': '1536kB', 'L3': '15MB',
-                          'memory': '1200MB'},
-        }
-
-        self.maintainers = ['SK', 'CB']
-        self.tags = {'benchmark', 'diagnostic', 'health'}
-
-        bw_pattern = sn.min(sn.extractall(r'MByte/s:\s*(?P<bw>\S+)',
-                                          self.stdout, 'bw', float))
-
-        self.sanity_patterns = sn.assert_ge(bw_pattern, 0.0)
-        self.perf_patterns = {
-            'bandwidth': bw_pattern
-        }
-
-
-@rfm.parameterized_test(*[[l, k] for l in ['L1', 'L2', 'L3']
+@rfm.simple_test
+class CPUBandwidth(MemBandwidthTest):
+    config = parameter([*[[l, k] for l in ['L1', 'L2', 'L3']
                           for k in ['load_avx', 'store_avx']],
                         ['memory', 'load_avx'],
-                        ['memory', 'store_mem_avx'])
-class CPUBandwidth(MemBandwidthTest):
-    def __init__(self, mem_level, kernel_name):
-        super().__init__()
-
-        self.descr = f'CPU <- {mem_level} {kernel_name} benchmark'
-        self.valid_systems = ['daint:mc', 'daint:gpu', 'dom:gpu', 'dom:mc']
-
-        # the kernel to run in likwid
-        self.kernel_name = kernel_name
-        self.mem_level = mem_level
-
-        self.refs = {
-            'mc':  {
-                'load_avx': {'L1': 5100000, 'L2': 2000000, 'L3': 900000,
-                             'memory': 130000},
-                'store_avx': {'L1': 2800000, 'L2': 900000, 'L3': 480000},
-                'store_mem_avx': {'memory': 85000},
-            },
-            'gpu': {
-                'load_avx': {'L1': 2100000, 'L2': 850000, 'L3': 360000,
-                             'memory': 65000},
-                'store_avx': {'L1': 1200000, 'L2': 340000, 'L3': 210000},
-                'store_mem_avx': {'memory': 42500},
-            }
+                        ['memory', 'store_mem_avx']])
+    valid_systems = ['daint:mc', 'daint:gpu', 'dom:gpu', 'dom:mc']
+    # the kernel to run in likwid
+    kernel_name = variable(str)
+    mem_level = variable(str)
+    refs = {
+        'mc':  {
+            'load_avx': {'L1': 5100000, 'L2': 2000000, 'L3': 900000,
+                         'memory': 130000},
+            'store_avx': {'L1': 2800000, 'L2': 900000, 'L3': 480000},
+            'store_mem_avx': {'memory': 85000},
+        },
+        'gpu': {
+            'load_avx': {'L1': 2100000, 'L2': 850000, 'L3': 360000,
+                         'memory': 65000},
+            'store_avx': {'L1': 1200000, 'L2': 340000, 'L3': 210000},
+            'store_mem_avx': {'memory': 42500},
         }
-        ref_proxy = {part: self.refs[part][kernel_name][mem_level]
-                     for part in self.refs.keys()}
+    }
 
+    @run_after('init')
+    def set_(self):
```

Comment:
change to `setup_descr` and remove the reference related stuff to a separate method